nixos/modules/services/networking/livekit.nix

```
@@ -0,0 +1,130 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.livekit;
+  format = pkgs.formats.json { };
+in
+{
+  meta.maintainers = with lib.maintainers; [ quadradical ];
+  options.services.livekit = {
+    enable = lib.mkEnableOption "Enable the livekit server";
+    package = lib.mkPackageOption pkgs "livekit" { };
+
+    environmentFile = lib.mkOption {
+      type = lib.types.path;
+      description = ''
+        LiveKit key file, with syntax `LIVEKIT_KEYS=\"key: secret\"`
+        The key and secret are used by other clients or services to connect to your Livekit instance.
+      '';
+    };
```

```suggestion
    environmentFile = lib.mkOption {
      type = with lib.types; nullOr path;
      description = ''
        Path to a file with environment variables passed to the livekit-server process.
      '';
    };
```

We should use the `--key-file` parameter instead to pass the dedicated secret file in and change this option to become an escape hatch.

We should then use the `LoadCredential=` interface of systemd to safely provide the secret file to the systemd unit.

```nix
  LoadCredential = [ "livekit-secrets:${cfg.keyFile}" ];
  ExecStart = utils.escapeSystemdExecArgs ([
    # abbreviated.
    "--keys-file=/run/credentials/livekit.service/livekit-secrets"
  ] ...);
```

Normally we would use `$CREDENTIALS_DIRECTORY`, but `escapeSystemdExecArgs` escapes that, so we are using the path directly like that.

The option description should reflect how to generate the key file, namely:

```shell
$ livekit-server generate-keys
API Key:  APIvVUvhaMt8y7m
API Secret:  Fsk5UzbapJgbqej3evrjoVZl7Xor1ZiAXi2sTxpCS7o
```

---

Huh, I just created my keys manually, I didn't know there was a generator.

---

Fixed.